packages/netlify/src/image-service.ts

```diff
@@ -17,9 +17,18 @@ const service: ExternalImageService = {
 	getURL(options) {
 		const query = new URLSearchParams();
 
-		const fileSrc = isESMImportedImage(options.src)
+		let fileSrc = isESMImportedImage(options.src)
 			? removeLeadingForwardSlash(options.src.src)
 			: options.src;
+		const site = import.meta.env.SITE;
+		if (URL.canParse(fileSrc)) {
+			const url = new URL(fileSrc);
+			if (import.meta.env.NETLIFY_LOCAL && url.hostname === 'localhost') {
```

Comment:
Would this work if the user starts the astro dev server with `--host`? This option exposes the server to the local network, so `localhost` isn't valid anymore

Reply:
It only works if you're running it using the Netlify CLI (which is what the first env var check is), which wouldn't work with `--host`